music_assistant/server/providers/radiobrowser/__init__.py

```diff
@@ -179,21 +174,20 @@ async def browse(self, path: str, offset: int, limit: int) -> list[MediaItemType
                     )
                 ]
                 items.append(folder)
-            return items
-
-        if subsubpath in await self.get_tag_names(limit=limit, offset=offset):
-            return await self.get_by_tag(subsubpath)
+            return items[offset:(offset+limit)]
```

Comment:
this bad - this fetches all data and then truncates it - then you can better leave out all paging completely

Reply:
Unfortunately the countries method doesn't have a filter so I thought to leave the paging by slicing, but happy to remove it and get it all in once. Does the method need to add a cache decorator or it pass that over from the main method ?

Reply:
there doesn't need to be a filter - the filter is in the frontend - you just need to return a sane amount of chunked data - the frontend will notice that the endpoint doesn't do filtering and grab all items first. For lists that have < 1000 items its fine to just return them all at once and save the paging overhead.

What version did you test this on? I think all this logic is fixed in the recent betas of MA but might be still broken on the stable version.

Reply:
Latest version. 
The issue is that we have like 200+ countries/tags and the frontend is asking :
"look for X in  offset 0 limit 50", then the rest is not looked up. That is why the ticket above is not showing GB radios

Reply:
Nope, that is not how it works. Are you on latest beta or stable ?

Reply:
I was on the main branch, which I suppose is the stable one? 
I synced to the source main and I just worked from there, let me know if I should work on another base of the branch.
I added logs and that what was happening on my side.

Reply:
I just tested and its working fine on my end - hence my question how you tested this